spec/models/events/steps/personalised_updates_spec.rb

```diff
@@ -0,0 +1,68 @@
+require "rails_helper"
+
+describe Events::Steps::PersonalisedUpdates do
+  include_context "wizard step"
+  include_context "stub types api"
+
+  it_behaves_like "a wizard step"
+
+  context "attributes" do
+    it { is_expected.to respond_to :address_postcode }
+    it { is_expected.to respond_to :degree_status_id }
+    it { is_expected.to respond_to :consideration_journey_stage_id }
+    it { is_expected.to respond_to :preferred_teaching_subject_id }
+  end
+
+  context "validations" do
+    it { is_expected.to allow_value("TE571NG").for :address_postcode }
+    it { is_expected.to allow_value("TE57 1NG").for :address_postcode }
+    it { is_expected.to allow_value(" TE57 1NG ").for :address_postcode }
+    it { is_expected.to allow_value("").for :address_postcode }
+    it { is_expected.not_to allow_value("unknown").for :address_postcode }
+
+    it { is_expected.to allow_value(subject.degree_status_option_ids.first).for :degree_status_id }
+    it { is_expected.to allow_value(subject.degree_status_option_ids.last).for :degree_status_id }
+    it { is_expected.not_to allow_value(nil).for :degree_status_id }
+    it { is_expected.not_to allow_value("12345").for :degree_status_id }
+
+    it { is_expected.to allow_value(subject.journey_stage_option_ids.first).for :consideration_journey_stage_id }
+    it { is_expected.to allow_value(subject.journey_stage_option_ids.last).for :consideration_journey_stage_id }
+    it { is_expected.not_to allow_value(nil).for :consideration_journey_stage_id }
+    it { is_expected.not_to allow_value("12345").for :consideration_journey_stage_id }
+
+    it { is_expected.to allow_value(subject.teaching_subject_option_ids.first).for :preferred_teaching_subject_id }
+    it { is_expected.to allow_value(subject.teaching_subject_option_ids.last).for :preferred_teaching_subject_id }
+    it { is_expected.not_to allow_value(nil).for :preferred_teaching_subject_id }
+    it { is_expected.not_to allow_value("12345").for :preferred_teaching_subject_id }
+  end
+
+  context "data cleaning" do
+    it "cleans the postcode" do
```

Comment:
One for later, but I wrote a small shared example to deal with this for the TTA app that we could pull in here as well:

https://github.com/DFE-Digital/get-teacher-training-adviser-service/blob/master/spec/support/shared_examples/sanitize_input.rb

Then this becomes:

```
include_context "sanitize fields", %i[address_postcode]
```